library/pkparse.c

```diff
@@ -1360,7 +1384,12 @@ int mbedtls_pk_parse_public_key( mbedtls_pk_context *ctx,
 #if defined(MBEDTLS_PEM_PARSE_C)
     size_t len;
     mbedtls_pem_context pem;
+#endif
+
+    PK_VALIDATE_RET( ctx != NULL );
+    PK_VALIDATE_RET( key != NULL || keylen == 0 );
```

Comment:
Depending on the library configuration, this function checks for `keylen == 0` later on and rejects it. Do we need another check here? Could we remove the latter check?

Reply:
I've refactored the `keylen==0` checks to make the behavior more obvious: reject `keylen==0` early and unconditionally instead of testing it multiple times later.